### ISSUE TITLE
Enhance a11y for `Table`

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#5987]
name: Table

### Description
0 accessibility violations were found in the `Table`.

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.
